packages/ember-metal/lib/mixin.js

```diff
@@ -135,19 +135,7 @@ function giveMethodSuper(obj, key, method, values, descs) {
 
 function applyConcatenatedProperties(obj, key, value, values) {
   let baseValue = values[key] || obj[key];
-  let ret;
-
-  if (baseValue === null || baseValue === undefined) {
-    ret = makeArray(value);
-  } else if (isArray(baseValue)) {
-    if (value === null || value === undefined) {
-      ret = baseValue;
-    } else {
-      ret = a_concat.call(baseValue, value);
-    }
-  } else {
-    ret = a_concat.call(makeArray(baseValue), value);
-  }
+  let ret = makeArray(baseValue).concat(makeArray(value));
```

Comment:
`makeArray` already has `null/undefined` checks which was done in `applyConcatenatedProperties` before

Reply:
there was a performance gain from avoiding the concat when it wasn't needed, this would need to be retested